rest-service/manager_rest/utils.py

```diff
@@ -411,17 +411,47 @@ def get_rrule(rule, since, until):
     if rule.get('weekdays'):
         weekdays = _get_weekdays(rule['weekdays'])
 
-    return rrule.rrule(freq=frequency, interval=interval,
-                       dtstart=since, until=until, byweekday=weekdays,
-                       count=rule.get('count'), cache=True)
-
-
-def _get_weekdays(weekdays):
-    weekdays_rule = []
-    for weekday in rrule.weekdays:
-        if str(weekday) in weekdays:
-            weekdays_rule.append(weekday)
-    return weekdays_rule
+    if not weekdays:
+        return rrule.rrule(freq=frequency, interval=interval,
+                           dtstart=since, until=until, count=rule.get('count'),
+                           cache=True)
+
+    rule_set = rrule.rruleset()
+    count = rule.get('count')
+    for weekday in weekdays:
+        rule_set.rrule(
+            rrule.rrule(freq=frequency, interval=interval, dtstart=since,
+                        until=until, byweekday=weekday[0], bysetpos=weekday[1],
+                        cache=True))
+    if not count:
+        return rule_set
+    try:
+        until_cap = rule_set[count-1]
+    except IndexError:
+        return rule_set
+    capped_rule_set = rrule.rruleset()
+    for weekday in weekdays:
+        capped_rule_set.rrule(
+            rrule.rrule(freq=frequency, interval=interval, dtstart=since,
+                        until=until_cap, byweekday=weekday[0],
+                        bysetpos=weekday[1], cache=True))
+    return capped_rule_set
+
+
+def _get_weekdays(raw_weekdays):
```

Comment:
as usual, this needs at least a docstring, because it's impossible to understand without reading the whole function, what is the argument and return values